plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py

```diff
@@ -701,20 +664,21 @@ def get_deploy_transaction(
             if kwargs.get("chain_id", None) is not None
             else self.chain_id
         )
-        account_number, sequence = self._try_get_account_number_and_sequence(
-            deployer_address
-        )
+
+        account_number = kwargs.pop("account_number", None)
+        sequence = kwargs.pop("sequence", None)
+
         if account_number is None or sequence is None:
-            return None  # pragma: nocover
+            account_number, sequence = self._try_get_account_number_and_sequence(
+                deployer_address
+            )
+            if account_number is None or sequence is None:
+                return None  # pragma: nocover
+
         label = kwargs.pop("label", None)
         code_id = kwargs.pop("code_id", None)
         amount = kwargs.pop("amount", None)
         init_msg = kwargs.pop("init_msg", None)
-        unexpected_keys = [
-            key for key in kwargs.keys() if key not in ["tx_fee", "gas", "memo"]
```

Comment:
Why not keep this with the added `account_number` and `sequence`?

Reply:
These variables can be part of any message same for account_number and sequence hence they are not invalid.

Reply:
Put back